### PR TITLE
migrate test_chroot to pytest

### PR DESCRIPTION
**Pull request by frogunder**
### What does this PR do?

migrate test_chroot to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by MKLeb — DISMISSED** [review]

File: tests/pytests/unit/modules/test_chroot.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+import salt.loader.context
+import salt.modules.chroot as chroot
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+from tests.support.unit import skipIf
+
+
+@skipIf(salt.utils.platform.is_windows(), "This test cannot work on Windows")
```
Comment (MKLeb) [inline tests/pytests/unit/modules/test_chroot.py:40]:
```suggestion
@pytest.mark.skipif(salt.utils.platform.is_windows(), reason="This test cannot work on Windows")
```

**Comment by MKLeb** [comment]
@frogunder These tests were previously skipped on windows.  I think we should make sure the migrated version is skipped on Windows as well.

File: tests/pytests/unit/modules/test_chroot.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+import salt.loader.context
+import salt.modules.chroot as chroot
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.mark.skipif(
+    salt.utils.platform.is_windows(), reason="This test cannot work on Windows"
+)
```
Comment (MKLeb) [inline tests/pytests/unit/modules/test_chroot.py:39]:
```suggestion
pytestmark = [
    pytest.mark.skipif(
        salt.utils.platform.is_windows(), reason="This test cannot work on Windows"
    )
]
```

This should add a global skip.

File: tests/pytests/unit/modules/test_chroot.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
+#
```
Comment (Ch3LL) [inline tests/pytests/unit/modules/test_chroot.py:1]: please remove all of these comments. line 1-19

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_chroot.py (hunk 3 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_chroot.py]: removed

File: tests/pytests/unit/modules/test_chroot.py (hunk 2 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_chroot.py]: thanks

**Review by MKLeb — DISMISSED** [review]

File: tests/pytests/unit/modules/test_chroot.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+@patch("os.path.isdir")
+def test_exist(isdir):
+    """
+    Test if the chroot environment exist.
+    """
+    isdir.side_effect = (True, True, True, True)
+    assert chroot.exist("/chroot")
+
+    isdir.side_effect = (True, True, True, False)
+    assert not chroot.exist("/chroot")
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_chroot.py:55]:
```suggestion
def test_exist():
    """
    Test if the chroot environment exist.
    """
    with patch("os.path.isdir") as isidr:
        isdir.side_effect = (True, True, True, True)
        assert chroot.exist("/chroot")

    with patch("os.path.isdir") as isidr:
        isdir.side_effect = (True, True, True, False)
        assert not chroot.exist("/chroot")
```
And the same for all `@patch` decorators please

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_chroot.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+import salt.modules.chroot as chroot
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+
+pytestmark = [
+    pytest.mark.skipif(
+        salt.utils.platform.is_windows(), reason="This test cannot work on Windows"
+    )
+]
```
[the remark this answers is not in the document]
Comment (frogunder) [reply tests/pytests/unit/modules/test_chroot.py]: thanks

File: tests/pytests/unit/modules/test_chroot.py (hunk 4 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_chroot.py]: Thanks Pedro

**Review by Ch3LL — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by MKLeb — APPROVED** [review]

**Pull request merged**